### ISSUE TITLE
alternative triplet loss

### ISSUE THREAD

**Issue by eladhoffer** [adambielski/siamese-triplet#1]
might be interesting to compare with the alternative triplet loss described in: https://arxiv.org/abs/1412.6622

**Comment by adambielski** [comment]
Thanks! I've heard about this definition too - it might be interesting to compare. Might add it to the repo at some point.

**Comment by adambielski** [comment]
@preesee You can use whatever network definition in PyTorch you want. Just use it with appropriate loss functions.

**Comment by preesee** [comment]
I am trying to leverage triplet loss ,as some papers proves that triplet loss improve the accuracy.But triplet returns only (0,1) ,which means only negative or positive returns for the prediction values, correct? text similarity is a scale with values in range [0,5],do you think it is possible to measure similarity  of two pics into value in range , eg: [0,5] with triplet loss base on your framework ?

**Comment by adambielski** [comment]
I don't really know what problem exactly you're referring to. To apply it to a different problem than same/different within a class label, the triplet loss probably should be modified.

**Comment by preesee** [comment]
Hi adambielski,
I still have a question about Triplet network here: 
after the Triplet network model trained , how we can use the model in product environment then ?We still need to feed 3pics as a tuple to the network ? eg: let's suppose I have 3 pic[face1]:A, [face2]:P,[face3]N but I don't know if they are same person or not each other, how to feed this 3pics? If I already know face1 and face2 are in the same class/person and face3 is from a different person , what is the meaning for this model? A little confused about the usage of this model.
